runtime/src/bank.rs

```diff
@@ -822,6 +856,12 @@ impl Bank {
 
         if *hash == Hash::default() {
             // finish up any deferred changes to account state
+
+            // DANGER: Remove this guard after tds has transitioned to the eager rent
+            // collection and ABSOLUTELY before the mainnet-beta transitions to v1.1.
+            if !(self.operating_mode() == OperatingMode::Preview && self.epoch() < 45) {
```

Comment:
@mvines For ops perspective, this is the crucial gate for tds targeted at around May 17 (weekend, but this is tds) ec1f84f7ff190e358cccc86c75167348934e4bc9 :

```
$ ../solana-cli  --url http://tds.solana.com:8899 epoch-info

Slot: 13138639
Epoch: 43
Epoch slot range: [13052256..13484256)
Epoch completed percent: 19.996%
Epoch completed slots: 86383/432000 (345617 remaining)
Epoch completed time: 9h 35m 53s/2days (1day 14h 24m 6s remaining)

==> 2020-05-14 01:19:27.407609790|0
ryoqun@ubuqun:~/work/solana/solana$ pry
[2] pry(main)> Time.now + (24 * 2 + 24 + 14) * 3600
=> 2020-05-17 15:21:07 +0900
```

Reply:
Ah, also, if releasing in this time-frame is too tight, I'll happily adjust the activation epoch. :)

Reply:
I'd feel better about mid-late next week, so we can ask TdS validators to upgrade to new release next Monday

Reply:
Ok, I've rescheduled to be epoch **47**:

```
$ ../solana-cli  --url http://tds.solana.com:8899 epoch-info

Slot: 13327367
Epoch: 43
Epoch slot range: [13052256..13484256)
Epoch completed percent: 63.683%
Epoch completed slots: 275111/432000 (156889 remaining)
Epoch completed time: 1day 6h 34m 4s/2days (17h 25m 55s remaining)

==> 2020-05-14 23:45:53.771498477|0
$ cal
      May 2020        
Su Mo Tu We Th Fr Sa  
                1  2  
 3  4  5  6  7  8  9  
10 11 12 13 14 15 16  
17 18 19 20 21 22 23  
24 25 26 27 28 29 30  
31                    
$ pry
[14] pry(main)> Time.now + (156889 / 2.5  + (3600 * 24 * 2) * 3)
=> 2020-05-21 17:18:42 +0900

```

Reply:
Yep looks great.   

For mainnet-beta, backporting this to 1.0 is too risky but we'll also need similar gating logic for the Stable operating mode though.   How about adding the code for this gating into this PR but set the Stable epoch to u64::MAX, and then we can track reducing that value as a new item in https://github.com/solana-labs/solana/projects/54

Reply:
Hm, that sounds ok to me. So, we'll enable eager-rent-collection on mainnet-bete after while the mainnet-beta transitions to the v1.1?

FYI: @sakridge

Reply:
Yep, mainnet-beta moves to v1.1 at the end of the month so we're almost there anyway

Reply:
done!: https://github.com/solana-labs/solana/pull/10028/commits/5e5ae93860c68cbcb739bbf3d60a3c1230cec5d2